sponsorship_compassion/model/contracts.py

```diff
@@ -61,8 +67,43 @@ def _get_contract_from_invoice(invoice_obj, cr, uid, invoice_ids,
                         contract.write({
                             'activation_date': datetime.today().strftime(DF)})
 
+                        # Cancel the invoices if a contract is activated
+                        self._cancel_invoices(
+                            cr, uid,
+                            invoice.partner_id.id,
+                            first_pay_date,
+                            context)
         return list(res)
 
+    def _cancel_invoices(
+            self, cr, uid, partner_id, date_invoice, context=None):
+        invoice_obj = self.pool.get('account.invoice')
+        invoice_ids = invoice_obj.search(
+            cr, uid,
+            [('partner_id', '=', partner_id),
+             ('state', '=', 'open'),
+             ('date_invoice', '<', date_invoice)
+             ],
+            context=context)
+
+        for invoice in invoice_obj.browse(cr, uid, invoice_ids, context):
+            invoice_lines = invoice.invoice_line
+            contract_ids = [
+                invoice_line.contract_id.id for invoice_line in invoice_lines
+                if invoice_line.contract_id]
+            contract_ids = list(set(contract_ids))
+
+            wf_service = netsvc.LocalService('workflow')
+            if contract_ids:
+                if len(contract_ids) == 1:
+                    wf_service.trg_validate(uid, 'account.invoice',
+                                            invoice.id, 'invoice_cancel', cr)
+                else:
+                    invoice_obj.action_cancel_draft(
+                        cr, uid, invoice.id, context)
+                    invoice_obj.trg_validate(uid, 'account.invoice',

```

Comment:
Before validating again the invoice, you should remove the invoice line from the invoice (otherwise you are only cancelling and validating an invoice, which has no effect).